tests/unit/test_schema_reader.py

```diff
@@ -0,0 +1,44 @@
+from concurrent.futures import ThreadPoolExecutor
```

Comment:
I'm using a `ThreadPoolExecutor` instead of plain threads because the `Future` object does the work of capturing the exception and re-raising it on the main thread.